pkg/image/registry/imagestreamtag/rest.go

```diff
@@ -237,10 +240,16 @@ func (r *REST) Update(ctx apirequest.Context, tagName string, objInfo rest.Updat
 		if err := rest.BeforeCreate(Strategy, ctx, obj); err != nil {
 			return nil, false, err
 		}
+		if err := createValidation(obj.DeepCopyObject()); err != nil {
+			return nil, false, err
+		}
 	} else {
 		if err := rest.BeforeUpdate(Strategy, ctx, obj, old); err != nil {
 			return nil, false, err
 		}
+		if err := updateValidation(obj.DeepCopyObject(), old.DeepCopyObject()); err != nil {
```

Comment:
do we know old is nil? Aren't there ways to update without old object? Remembering a comment like that from @lavalamp in a different context. @liggitt ?

Reply:
> do we know old is nil? Aren't there ways to update without old object? Remembering a comment like that from @lavalamp in a different context. @liggitt ?

An update that allows create if it doesn't exist.  imagestreamtags don't do that.  Also, nil receivers are allowed and as I recall handled by the generated deep copy.

Reply:
nil receivers for struct pointers. For interfaces they panic.